CSharpSourceCode/Battle/CrosshairMissionBehavior/CustomCrosshairMissionBehavior.cs

```diff
@@ -140,18 +140,37 @@ private void UpdateAbilityCrosshairVisibility()
             {
                 if (!abilityComponent.CurrentAbility.CanCast(Agent.Main))
                 {
-                    if (abilityCrosshair.IsVisible)
-                        abilityCrosshair.Hide();
+                    abilityCrosshair.Hide();
                 }
                 else
                 {
-                    abilityCrosshair.Tick();
-                    if (!abilityCrosshair.IsVisible)
+                    if (abilityCrosshair.CrosshairType == CrosshairType.CenteredAOE || IsCorrectDirection())
+                    {
+                        abilityCrosshair.Tick();
                         abilityCrosshair.Show();
+                    }
+                    else
+                    {
+                        abilityCrosshair.Hide();
+                    }
                 }
             }
         }
 
+        private bool IsCorrectDirection()
```

Comment:
the name can be misleading. I would call it IsInFrontOfCharacter)

Reply:
Yeah, i wanted to call it the same, but actually player can cast ability to a side, anyway i'll think another name up